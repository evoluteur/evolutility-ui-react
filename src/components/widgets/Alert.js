import React from "react";
import PropTypes from "prop-types";

import "./Alert.scss";

const icons = {
  info: "info-sign",
  success: "ok-sign",
  warning: "warning-sign",
  danger: "exclamation-sign", //'remove-sign'
};
function icon(name) {
  return <i className={"glyphicon glyphicon-" + icons[name]} />;
}

const Alert = ({ title, message, type }) => (
  <div className={"alert alert-" + type} role="alert">
    {title ? icon(type) : null}
    <strong>{title}</strong>
    <p>{message}</p>
  </div>
);

export default Alert;

Alert.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    "info", // - blue
    "success", // - green
    "warning", // - yellow
    "danger", // - red
  ]),
};

Alert.defaultProps = {
  type: "danger",
};
