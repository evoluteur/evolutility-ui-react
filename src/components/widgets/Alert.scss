@import "../../variables";

.alert {
  position: relative;
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid rgba(0, 0, 0, 0);
  border-radius: $radius;
  > .alert-title {
    position: relative;
    margin: 0 0 20px;
    font-size: 22px;
    > svg {
      position: relative;
      top: 4px;
      margin-right: 7px;
      width: 24px;
      height: 24px;
    }
  }
  .alert-link {
    font-weight: bold;
  }
  > p,
  > ul {
    margin-bottom: 0;
    &:first-letter {
      text-transform: uppercase;
    }
  }
  > p + p {
    margin-top: 5px;
  }
}

.alert-success {
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
  .alert-link {
    color: #2b542c;
  }
  > .alert-title > svg {
    fill: #2b542c;
  }
}
.alert-info {
  color: #31708f;
  background-color: #d9edf7;
  border-color: #bce8f1;
  .alert-link {
    color: #245269;
  }
  > .alert-title > svg {
    fill: #245269;
  }
}
.alert-warning {
  color: #8a6d3b;
  background-color: #fcf8e3;
  border-color: #faebcc;
  .alert-link {
    color: #66512c;
  }
  > .alert-title > svg {
    fill: #66512c;
  }
}
.alert-danger {
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
  .alert-link {
    color: #843534;
  }
  > .alert-title > svg {
    fill: #843534;
  }
}
