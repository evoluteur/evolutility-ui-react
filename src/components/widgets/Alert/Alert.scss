// copied and modified from bootstrap-sass v3.4.1

@import "../../../variables";

.alert {
  position: relative;
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid silver;
  border-left-width: 12px;
  border-radius: $radius;
  > .alert-title {
    position: relative;
    margin: 0 0 20px;
    font-size: 22px;
    > svg {
      position: relative;
      top: 4px;
      margin-right: 7px;
      width: 24px;
      height: 24px;
    }
  }
  a {
    text-decoration: underline;
  }
  > div,
  > ul {
    margin-bottom: 0;
    &:first-letter {
      text-transform: uppercase;
    }
  }
}

// the following are shared b/w Alert and Badge components
.alert-success {
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
  a {
    color: #3c763d;
  }
  > .alert-title > svg {
    fill: #3c763d;
  }
}
.alert-info {
  color: #31708f;
  background-color: #d9edf7;
  border-color: #bce8f1;
  a {
    color: #31708f;
  }
  > .alert-title > svg {
    fill: #31708f;
  }
}
.alert-warning {
  color: #8a6d3b;
  background-color: #fcf8e3;
  border-color: #faebcc;
  a {
    color: #8a6d3b;
  }
  > .alert-title > svg {
    fill: #8a6d3b;
  }
}
.alert-danger {
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
  a {
    color: #a94442;
  }
  > .alert-title > svg {
    fill: #a94442;
  }
}
