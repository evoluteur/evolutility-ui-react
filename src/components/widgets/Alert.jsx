import React from "react";
import PropTypes from "prop-types";
import Icon from "react-crud-icons";

import "./Alert.scss";

const icons = {
  info: "info",
  success: "check",
  warning: "alert",
  danger: "error",
};

const icon = (name) => <Icon name={icons[name]} size="medium" theme="none" />;

const Alert = ({ title, message, type }) => (
  <div className={"alert alert-" + type} role="alert" data-testid="alert-test">
    <h3 className="alert-title">
      {title && icon(type)}
      <strong>{title}</strong>
    </h3>
    <p>{message}</p>
  </div>
);

export default Alert;

Alert.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    "info", // - blue
    "success", // - green
    "warning", // - yellow
    "danger", // - red
  ]),
};

Alert.defaultProps = {
  title: null,
  type: "danger",
};
