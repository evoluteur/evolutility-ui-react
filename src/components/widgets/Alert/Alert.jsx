import React, { memo } from "react";
import PropTypes from "prop-types";
import Icon from "react-crud-icons";

import "./Alert.scss";

const icons = {
  info: "info",
  success: "check",
  warning: "alert",
  danger: "error",
};

const icon = (name) => <Icon name={icons[name]} size="medium" theme="none" />;

const Alert = memo(({ title, message, type }) => (
  <div className={"alert alert-" + type} role="alert" data-testid="alert">
    <h3 className="alert-title">
      {title && icon(type)}
      <strong>{title}</strong>
    </h3>
    <div>{message}</div>
  </div>
));

export default Alert;

Alert.propTypes = {
  title: PropTypes.string,
  message: PropTypes.node.isRequired,
  type: PropTypes.oneOf([
    "info", // - blue
    "success", // - green
    "warning", // - yellow
    "danger", // - red
  ]),
};

Alert.defaultProps = {
  title: null,
  type: "danger",
};
