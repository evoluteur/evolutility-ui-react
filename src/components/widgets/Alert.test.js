/* eslint-disable import/no-extraneous-dependencies */
import { getAllByTitle, getByTitle, getByText, render, screen, TestRenderer } from "@testing-library/react";
import Alert from "./Alert";
import renderer from 'react-test-renderer';
import { Fragment } from "react";
import { isExportDeclaration } from "typescript";

describe('alert widget tests', () => {
  it('component renders correctly', () =>{;
    const tree = renderer.create(<Alert title="Alert" message = "hello" type="danger"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("alert has title and message", () => {
     render(<Alert title ="Hello!" message="you have alert" type="danger" />);
     const alert = screen.getByTestId("alert-test");
     expect(alert).toHaveTextContent("Hello!");
     expect(alert).toHaveTextContent("you have alert");
   });
   it("alert has null title and message", () => {
    render(<Alert title ="" message="you have alert" type="danger" />);
    const alert = screen.getByTestId("alert-test");
    expect(alert).toHaveTextContent("you have alert");
  });
  it("test type: info", () => {
    render(<Alert title ="" message="info alert" type="info" />);
    const alert = screen.getByTestId("alert-test");
    expect(alert).toHaveTextContent("info alert");
  });
  it("test type: success", () => {
    render(<Alert title ="" message="success alert" type="success" />);
    const alert = screen.getByTestId("alert-test");
    expect(alert).toHaveTextContent("success alert");
  });
  it("test type: warning", () => {
    render(<Alert title ="" message="warning alert" type="warning" />);
    const alert = screen.getByTestId("alert-test");
    expect(alert).toHaveTextContent("warning alert");
  });
})

