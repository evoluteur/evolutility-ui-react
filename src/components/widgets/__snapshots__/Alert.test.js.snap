// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert widget tests component renders correctly 1`] = `
<div
  className="alert alert-danger"
  data-testid="alert-test"
  role="alert"
>
  <h3
    className="alert-title"
  >
    <svg
      aria-hidden="true"
      className={null}
      focusable="false"
      onClick={null}
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M2.2,16.06L3.88,12L2.2,7.94L6.26,6.26L7.94,2.2L12,3.88L16.06,2.2L17.74,6.26L21.8,7.94L20.12,12L21.8,16.06L17.74,17.74L16.06,21.8L12,20.12L7.94,21.8L6.26,17.74L2.2,16.06M13,17V15H11V17H13M13,13V7H11V13H13Z"
      />
    </svg>
    <strong>
      Alert
    </strong>
  </h3>
  <p>
    hello
  </p>
</div>
`;
